lib/api/route/messages.dart

```diff
@@ -263,3 +263,60 @@ class UploadFileResult {
 
   Map<String, dynamic> toJson() => _$UploadFileResultToJson(this);
 }
+
+/// https://zulip.com/api/add-reaction
+Future<AddReactionResult> addReaction(ApiConnection connection, {
+  required int messageId,
+  required ReactionType reactionType,
+  required String emojiCode,
+  required String emojiName,
+}) {
+  return connection.post('addReaction', AddReactionResult.fromJson, 'messages/$messageId/reactions', {
+    'emoji_name': RawParameter(emojiName),
+    'emoji_code': RawParameter(emojiCode),
+    'reaction_type': RawParameter(switch (reactionType) {
+      ReactionType.unicodeEmoji => 'unicode_emoji',
+      ReactionType.realmEmoji => 'realm_emoji',
+      ReactionType.zulipExtraEmoji => 'zulip_extra_emoji',
+    }),
+  });
+}
+
+@JsonSerializable(fieldRename: FieldRename.snake)
+class AddReactionResult {
+  AddReactionResult();
```

Comment:
Hmm, this class doesn't really do anything, does it :-)

I think when there's no content in the response, it's probably cleanest to have the route binding just return `Future<void>`. The `fromJson` callback can look like `(_) {}`.